### PR TITLE
community[minor]: feat: Add Zep Cloud community components, update docs and examples

### PR DESCRIPTION
**Pull request by paul-paliychuk**
We have recently released our new `zep-cloud` sdks that are compatible with Zep Cloud (not Zep Open Source). 

We have also maintained our Cloud version of langchain components (ChatMessageHistory, VectorStore) as part of our sdks. 

This PRs goal is to port these components to langchain community repo, and close the gap with the existing Zep Open Source components already present in community repo (added `ZepCloudMemory`,`ZepCloudVectorStore`,`ZepCloudRetriever`). 
Also added a `ZepCloudChatMessageHistory` components together with some expression language examples ported from our repo. 

Also includes the documentation update - added pages for Zep Cloud components.

We have left the original open source components intact on purpose as to not introduce any breaking changes.

### CODE REVIEW

File: libs/langchain-community/src/vectorstores/zep_cloud.ts [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+   */
+  private async createCollection(args: IZepCloudConfig) {
+    await this.client.document.addCollection(args.collectionName, {
+      description: args.description,
+      metadata: args.metadata,
+    });
+  }
+
+  /**
+   * @deprecated Adding vectors in unsupported in Zep Cloud
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/zep_cloud.ts:131]: Deprecated is the wrong framing - just throw an error

File: libs/langchain-community/src/vectorstores/zep_cloud.ts [hunk 2]
```diff
@@ -0,0 +1,347 @@
+import { ZepClient } from "@getzep/zep-cloud";
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/zep_cloud.ts:1]: I don't think this makes sense as a vectorstore - why not just use the retriever interface?

**Comment by jacoblee93** [comment]
Thanks for waiting! Please also run `yarn format`.

**Comment by paul-paliychuk** [comment]
@jacoblee93 my bad, just ran the formatter

File: libs/langchain-community/src/vectorstores/zep_cloud.ts (hunk 2 above)
Comment (paul-paliychuk) [reply libs/langchain-community/src/vectorstores/zep_cloud.ts:1]: So the retriever is used for memory (if you want to search for messages or summaries from the conversation). The VectorStore is a class used to interact with zep collections that you can add documents to (not conversations), and then search those collections for rag use cases in the chains.

Retriever searches memory, Vector Store adds documents to collections and searches those documents. 

We have a Zep Vector Store in Python langchain sdk as well.

File: libs/langchain-community/src/vectorstores/zep_cloud.ts (hunk 1 above)
Comment (paul-paliychuk) [reply libs/langchain-community/src/vectorstores/zep_cloud.ts]: Thanks, addressed

File: examples/src/guides/expression_language/zep/zep_cloud_vector_store.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+]);
+
+const model = new ChatOpenAI({
+  temperature: 0.8,
+  modelName: "gpt-3.5-turbo-1106",
+});
+const retriever = vectorStore.asRetriever();
+
+const setupAndRetrieval = RunnableMap.from({
+  context: new RunnableLambda({
```
Comment (jacoblee93) [inline examples/src/guides/expression_language/zep/zep_cloud_vector_store.ts:60]: No big deal, but you don't need to wrap in the `RunnableLambda` constructor. Raw functions will be coerced.

**Comment by jacoblee93** [comment]
Thank you! Couldn't push to your branch, so pushed updates here: #5542

**Comment by jacoblee93** [comment]
Closed in favor of #5542

**Pull request closed without merging**